src/pages/signin/LoginForm.js

```diff
@@ -36,26 +40,39 @@ const propTypes = {
 };
 
 const defaultProps = {
-    account: {},
+    account: {
+        login: '',
+    },
 };
 
 class LoginForm extends React.Component {
     constructor(props) {
         super(props);
 
+        this.updateLogin = this.updateLogin.bind(this);
         this.validateAndSubmitForm = this.validateAndSubmitForm.bind(this);
+        this.login = props.account.login;
 
         this.state = {
             formError: false,
-            login: '',
         };
     }
 
+    /**
+     * Update the value of login in Onyx
+     *
+     * @param {String} newLogin
+     */
+    updateLogin(newLogin) {
+        this.login = newLogin;
+        Onyx.merge(ONYXKEYS.ACCOUNT, {login: newLogin});
```

Comment:
Hmm this seems incorrect. We should not set Onyx values directly inside components. That's what the "actions" are for.